File: Anul II/Medii de proiectare si programare/Laborator/lab05/Ubb.BikeContest.Repository/src/IParticipantRepository.cs
using System.Collections;
using Ubb.BikeContest.Model;

namespace Ubb.BikeContest.Repository;

public interface IParticipantRepository : IRepository<long, Participant>
{
    IEnumerable<Participant> GetParticipantsByTeam(long teamId);

    Participant GetParticipantByData(Participant participant);
}

File: Anul II/Medii de proiectare si programare/Laborator/lab09/persistence/src/main/java/com/ubb/repository/IParticipantRepository.java
package com.ubb.repository;

import com.ubb.model.Participant;

import java.util.Collection;
import java.util.Optional;

public interface IParticipantRepository extends IRepository<Long, Participant> {
    Collection<Participant> getParticipantsByTeam(final Long teamID);

    Optional<Participant> getParticipantByData(final Participant participant);
}
